opta/core/aws.py

```diff
@@ -77,6 +78,23 @@ def delete_opta_config(self) -> None:
         if resp["ResponseMetadata"]["HTTPStatusCode"] != 204:
             raise Exception(f"Failed to delete opta config in {bucket}/{config_path}.")
 
+        s3_client = boto3.client("s3")
+        resp = s3_client.delete_object(Bucket=bucket, Key=self.layer.name)
+        if resp["ResponseMetadata"]["HTTPStatusCode"] != 204:
+            raise Exception(
+                f"Failed to delete opta config in {bucket}/{self.layer.name}."
+            )
+
+        providers = self.layer.gen_providers(0)
+        dynamodb_table = providers["terraform"]["backend"]["s3"]["dynamodb_table"]
+
+        dynamodb_client: DynamoDBClient = boto3.client(
+            "dynamodb", config=Config(region_name=self.region)
+        )
+        dynamodb_client.delete_item(
```

Comment:
Why not just delete the whole dynamodb table?

Reply:
Because the table has the locks for all other services in the environment and the environment as well. We only care about removing the lock for the current service or the environment (and if we're doing it for the environment then that means we would have already cleared the locks for the services and the next step would be deleting the fully empty table-- totally graceful)